mongo/integration/mtest/setup.go

```diff
@@ -300,19 +311,50 @@ func addCompressors(uri string) string {
 	return addOptions(uri, "compressors=", comp)
 }
 
+func addServerlessAuthCredentials(uri string) (string, error) {
+	if os.Getenv("SERVERLESS") != "serverless" {
+		return uri, nil
+	}
+	user := os.Getenv("SERVERLESS_ATLAS_USER")
+	if user == "" {
+		return "", fmt.Errorf("serverless expects SERVERLESS_ATLAS_USER to be set")
+	}
+	password := os.Getenv("SERVERLESS_ATLAS_PASSWORD")
+	if user == "" || password == "" {
+		return "", fmt.Errorf("serverless expects SERVERLESS_ATLAS_PASSWORD to be set")
+	}
+
+	var scheme string
+	// remove the scheme
+	if strings.HasPrefix(uri, "mongodb+srv://") {
+		scheme = "mongodb+srv://"
+	} else if strings.HasPrefix(uri, "mongodb://") {
```

Comment:
Is there a reason we need to assert specifically that the scheme is either `"mongodb+srv"` or `"mongodb"`?

Optional: Consider simplifying these checks by splitting on `"://"` instead of checking for prefixes.
E.g.
```go
parts := strings.SplitN(uri, "://", 2)
if len(parts) < 2 {
	return "", errors.New(`invalid URI: does not include "://"`)
}
return fmt.Sprintf("%s://%s:%s@%s", parts[0], user, password, parts[1]), nil
```

Optional 2: Manually parsing and modifying URIs is a huge pain. Consider refactoring some/all of this code to use `url.Parse`.
E.g.
```go
u, err := url.Parse(uri)
if err != nil {
	return "", err
}
u.User = url.UserPassword(user, password)
return u.String()
```

Reply:
Discussed offline in the ticket channel: we only need to support `mongodb` and `mongodb+srv` URI schemes for testing. Atlas only returns `mongodb://` URI schemes, but drivers-evergreen-tools also provides the SRV version in case a driver needs the seed list. 

If the URI scheme is not one of `mongodb://` or `mongodb+srv://`, something has probably gone wrong, so I'm inclined to keep this assertion.